app/src/main/java/com/berniesanders/fieldthebern/mortar/ParcelableScreen.java

```diff
@@ -0,0 +1,64 @@
+package com.berniesanders.fieldthebern.mortar;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.util.SparseArray;
+import java.lang.reflect.Array;
+
+/**
+ * Extends Screen for persistence.  Basically just takes care of persisting the view
+ * state, and has template methods so that subclasses can persist anything else they
+ * might need (generally parameters).
+ */
+public abstract class ParcelableScreen extends Screen implements Parcelable {
+  @Override public final int describeContents() {
+    return 0;
+  }
+
+  @Override public final void writeToParcel(Parcel parcel, int i) {
+    //noinspection unchecked
+    SparseArray<Object> sparseArray = (SparseArray) getViewState();
+    parcel.writeSparseArray(sparseArray);
+    doWriteToParcel(parcel, i);
+  }
+
+  protected void doWriteToParcel(Parcel parcel, int flags) {
+  }
+
+  public static <T extends ParcelableScreen> Creator<T> zeroArgsScreenCreator(
```

Comment:
I first got some traction on this by plugging in code lifted from here: https://github.com/square/flow/issues/11#issuecomment-50980051

But I forgot about this method, which can spare us some parceling boilerplate in the screens with empty constructors.  I'll apply it in the next round.